airflow-core/src/airflow/configuration.py

```diff
@@ -2301,9 +2288,8 @@ def ensure_secrets_loaded(
     """
     # Check if the secrets_backend_list contains only 2 default backends.
 
-    # Check if we are loading the backends for worker too by checking if the default_backends is equal
-    # to DEFAULT_SECRETS_SEARCH_PATH.
-    if len(secrets_backend_list) == 2 or default_backends != DEFAULT_SECRETS_SEARCH_PATH:
+    # Check if we are loading the backends for worker too by checking if the default_backends is not None
+    if len(secrets_backend_list) == 2 or default_backends is not None:
         return initialize_secrets_backends(default_backends=default_backends)
     return secrets_backend_list
```

Comment:
With: https://github.com/apache/airflow/pull/57744 merged, there is a shared config parser now. You will have to update this in sdk/configuration.py too